packages/backend/src/core/preprocessing/StateDetailsPreprocessor.ts

```diff
@@ -43,6 +48,56 @@ export class StateDetailsPreprocessor {
     )
   }
 
+  async catchUpL2Transactions(
+    trx: Knex.Transaction,
+    preprocessToStateUpdateId: number
+  ) {
+    const recordsToUpdate =
+      await this.preprocessedStateDetailsRepository.getAllWithoutL2TransactionStatisticsUpToStateUpdateId(
+        preprocessToStateUpdateId,
+        trx
+      )
+
+    for (const recordToUpdate of recordsToUpdate) {
+      this.logger.info(
+        `Catching up L2 transaction statistics for state update ${recordToUpdate.stateUpdateId}`
+      )
+
+      const statisticsForStateUpdate =
+        await this.l2TransactionRepository.getStatisticsByStateUpdateId(
+          recordToUpdate.stateUpdateId,
+          trx
+        )
+
+      const previousStateUpdateDetails =
+        await this.preprocessedStateDetailsRepository.findByStateUpdateId(
+          recordToUpdate.stateUpdateId - 1,
+          trx
+        )
+
+      if (recordToUpdate.stateUpdateId > 1 && !previousStateUpdateDetails) {
+        throw new Error(
+          'Something wrong went when catching up L2 transactions for state details!'
```

Comment:
Maybe change to more explicit: Statistics for previous state update where not preprocessed. This should never happen.